templates/profile.php

```diff
@@ -204,10 +206,15 @@ function pmpromd_profile_shortcode($atts, $content=null, $code="")
 	if(empty($user_id) && !empty($_REQUEST['pu']))
 	{
 		//Get the profile user
-		if(is_numeric($_REQUEST['pu']))
-			$pu = get_user_by('id', $_REQUEST['pu']);
-		else
-			$pu = get_user_by('slug', $_REQUEST['pu']);
+		$user_identifier = apply_filters( 'pmpromd_user_identifier', 'user_nicename' );

```

Comment:
This is used in multiple places, let's make a new function that has the hook in that and call that function here.